src/controllers/requests/requests.ts

```diff
@@ -0,0 +1,1062 @@
+
+import EmittableError from '../../classes/EmittableError'
+import { Session } from '../../classes/session'
+import SwapAndBridgeError from '../../classes/SwapAndBridgeError'
+import { ORIGINS_WHITELISTED_TO_ALL_ACCOUNTS } from '../../consts/dappCommunication'
+import { AccountId } from '../../interfaces/account'
+import { Banner } from '../../interfaces/banner'
+import { DappProviderRequest } from '../../interfaces/dapp'
+import { Network } from '../../interfaces/network'
+import { NotificationManager } from '../../interfaces/notification'
+import { BuildRequest } from '../../interfaces/requests'
+import {
+  SwapAndBridgeActiveRoute,
+  SwapAndBridgeSendTxRequest
+} from '../../interfaces/swapAndBridge'
+import { Calls, DappUserRequest, SignUserRequest, UserRequest } from '../../interfaces/userRequest'
+import { WindowManager } from '../../interfaces/window'
+import { isBasicAccount, isSmartAccount } from '../../libs/account/account'
+import { getBaseAccount } from '../../libs/account/getBaseAccount'
+import { Call } from '../../libs/accountOp/types'
+// eslint-disable-next-line import/no-cycle
+import {
+  dappRequestMethodToActionKind,
+  getAccountOpActionsByNetwork
+} from '../../libs/actions/actions'
+import { getAccountOpBanners } from '../../libs/banners/banners'
+import { getAmbirePaymasterService, getPaymasterService } from '../../libs/erc7677/erc7677'
+import { TokenResult } from '../../libs/portfolio'
+import {
+  ACCOUNT_SWITCH_USER_REQUEST,
+  buildSwitchAccountUserRequest,
+  makeAccountOpAction
+} from '../../libs/requests/requests'
+import { parse } from '../../libs/richJson/richJson'
+import {
+  buildSwapAndBridgeUserRequests,
+  getActiveRoutesForAccount
+} from '../../libs/swapAndBridge/swapAndBridge'
+import {
+  buildClaimWalletRequest,
+  buildMintVestingRequest,
+  buildTransferUserRequest
+} from '../../libs/transfer/userRequest'
+import { AccountsController } from '../accounts/accounts'
+// eslint-disable-next-line import/no-cycle
+import {
+  AccountOpAction,
+  Action,
+  ActionExecutionType,
+  ActionPosition,
+  ActionsController
+} from '../actions/actions'
+import { DappsController } from '../dapps/dapps'
+import EventEmitter, { Statuses } from '../eventEmitter/eventEmitter'
+import { KeystoreController } from '../keystore/keystore'
+import { NetworksController } from '../networks/networks'
+import { ProvidersController } from '../providers/providers'
+// eslint-disable-next-line import/no-cycle
+import { SelectedAccountController } from '../selectedAccount/selectedAccount'
+import { SignAccountOpController, SignAccountOpUpdateProps } from '../signAccountOp/signAccountOp'
+import { SwapAndBridgeController, SwapAndBridgeFormStatus } from '../swapAndBridge/swapAndBridge'
+import { TransferController } from '../transfer/transfer'
+
+const STATUS_WRAPPED_METHODS = {
+  buildSwapAndBridgeUserRequest: 'INITIAL'
+} as const
+
+export class RequestsController extends EventEmitter {
```

Comment:
Please add a few notes to give context on what this controller is responsible for. I think you can use the reasoning in https://github.com/AmbireTech/ambire-app/issues/5148